inc/osvr/Server/ConfigFilePaths.h

```diff
@@ -38,8 +39,17 @@
 namespace osvr {
 namespace server {
 
+    /// @brief Get list of directories to search for config files.
+    OSVR_SERVER_EXPORT std::vector<std::string> getConfigDirectories();
+
     /// @brief this returns a vector of default server configuration file paths.
     OSVR_SERVER_EXPORT std::vector<std::string> getDefaultConfigFilePaths();
-}
-}
+
+    /// @brief given a input filename/path, returns potential full paths from
+    /// combining relative input paths with potential config file locations.
+    OSVR_SERVER_EXPORT std::vector<std::string>
+    tryResolvingPath(std::string const &inputFilename);
```

Comment:
That sounds reasonable. I try not to check if files exist (instead just opening them) to avoid file existence race conditions. I know they're obscure, but this way I am careful to write clear, correct code. Feel free to push a change to this branch. (bonus points if you can figure out why newer CMake is parsing some script wrong and breaking Travis after I went and explicitly fixed it)

Reply:
I'll check into that.

Reply:
I pushed a commit that renames `tryResolvingPath()` to `getCandidateConfigFilePaths()` (and adjusted the local iter variable name accordingly). Feel free to accept/reject.